src/librustc_typeck/diagnostics.rs

```diff
@@ -3426,6 +3426,47 @@ parameters. You can read more about it in the API documentation:
 https://doc.rust-lang.org/std/marker/struct.PhantomData.html
 "##,
 
+E0436: r##"
+Functional update syntax can only be used on struct types.
+Erroneous code example:
+
+```compile_fail
+enum Foo { A { x: u32, y: u32 } }
+
+fn bar() {
```

Comment:
It seemed to me that it was important to show how to perform the equivalent operation on enums. I don't feel like the example solution is very good without showing how to do the `match` and it seemed awkward to do the `match` when the variable definition is on the previous line. I left `bar` here so that it would be easier to see the difference between this and the example solution. I can remove it though if you think it would be clearer without it.